keps/sig-scheduling/2891-simplified-config/README.md

```diff
@@ -0,0 +1,861 @@
+to initialize all `MultiPoint` plugins as all extension points (this is the section of 
+code that utilizes the factory function stored in the `frameworkImpl` struct)
+
+```go
+// initialize regular extension points (current behavior)
+for _, e := range f.getExtensionPoints(profile.Plugins) {
+  if err := updatePluginList(e.slicePtr, *e.plugins, pluginsMap); err != nil {
+    return nil, err
+  }
+}
+
+// initialize MultiPoint plugins
+for _, e := range f.getExtensionPoints(profile.Plugins) {
+  plugins := reflect.ValueOf(e.slicePtr).Elem()
+  pluginType := plugins.Type().Elem()
+  // build set of plugins already registered via normal extension points
+  // to check double registration
+  set := sets.NewString()
+  for _, plugin := range *e.plugins {
+    set.Insert(plugin.Name)
+  }
+  
+  for _, ep := range profile.Plugins.MultiPoint.Enabled {
+    pg, ok := pluginsMap[ep.Name]
+    if !reflect.TypeOf(pg).Implements(pluginType) {
+      continue
+    }
+    
+    if !ok {
+      return fmt.Errorf("%s %q does not exist", pluginType.Name(), ep.Name)
+    }
+    
+    if set.Has(ep.Name) {
+      return fmt.Errorf("plugin %q already registered as %q", ep.Name, pluginType.Name())
+    }
+    
+    set.Insert(ep.Name)
+    
+    newPlugins := reflect.Append(plugins, reflect.ValueOf(pg))
+    plugins.Set(newPlugins)
+  }
+}
+```
+
+This code is similar to the existing [`updatePluginList`](https://github.com/kubernetes/kubernetes/blob/dc079acc2be603fbc2065e2748c79bb5e2c3453d/pkg/scheduler/framework/runtime/framework.go#L418)
+function except for 2 notable differences:
+
+1. It does not return an error if a plugin doesn't implement an extension point, it just continues
+2. It is only looping through the MultiPoint config field for each internal extension point
+
+With this implementation, there is no option for plugins or users to define a subset of extension 
+points to enable or disable should we deprecate the existing regular extension points. This is because 
+the plugins themselves do not have control over when their functions are called, and we have cast them 
+into each extension point. At that point the framework, not the plugin, takes over.
+
+Disabling of individual subsets could be preserved using modified config semantics as below:
+
+|   |   |enabled: xyz|disabled: xyz|enabled: *|disabled: *|
+|---|---|------------|-------------|----------|-----------|
+|in-tree|regular|reorder plugins/set weight|disable plugin xyz|no-op|disable all plugins of the point|\
+|   |multi-point|enable plugin xyz for all extension points|disable plugin xyz for all extension points|no-op|disable all in-tree plugins for all extension points|
+|out-of-tree|regular|enable plugin xyz|disable plugin xyz for this extension point|no-op|disable all plugins of this extension point|
```

Comment:
I don't think the distinction is between in-tree and out-of-tree, but enabled-by-default and disabled-by-default.

(We have in-tree plugins that are disabled by default)